src/platform/auth.ts

```diff
@@ -344,10 +346,10 @@ export const findUser = (
 						},
 					],
 				},
-				$select: ['id', 'actor', 'username', 'password'],
+				$select: $select as Writable<typeof $select>,
```

Comment:
Is there a way to allow both arrays and read-only arrays in pinejs client to make this usage easier?

Reply:
Yea, that's probably the best way to do it. I will take a look :+1:

Reply:
@Page- I tried adding `| readonly string[]` or `| ReadonlyArray<string>` in pine client but that didn't work :(
TS gets completely crazy even after adding the above and regardless of whether we pass `$select` w/o casting or w/ `as readonly string[]` emits the following:
```
Type 'readonly ["id", "actor", "username", "password"]' is not assignable to type '{ [index: string]: "asc" | "desc"; }'.
Index signature is missing in type 'readonly ["id", "actor", "username", "password"]'.ts(2345)
```
After some testing, it looks like once it sees a readonly array it can't match it w/ `$select` any more and ends up trying it against the `[index: string]: undefined | ParameterAlias | string[] | Filter | Expand | OrderBy;` of the `ODataOptions`.
Hopefully this might get solved in a newer version.
Some alternatives that at work are:
* `$select: $select as readonly string[] as string[],`
* `$select: $select.slice(),`
* dropping the `as const` and manually typing `'id' | 'actor' | 'username' | 'password'` (but that's repeating almost the exact same text).

Give me a note on how you prefer to proceed on this.

Reply:
An alternative that I thought is to make `findUser()` accept an optional `$select` parameter which would have the current values as the default. The typings feel better imo and also allow us to omit props as needed for each case.